components/inputs/input-date.js

```diff
@@ -109,21 +223,74 @@ class InputDate extends LocalizeStaticMixin(RtlMixin(LitElement)) {
 		if (elem) elem.focus();
 	}
 
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldVal, prop) => {
+			if (prop === 'value') {
+				if (this.value) this._formattedValue = formatISODateInUserCalDescriptor(this.value);
+				else this._formattedValue = '';
+
+			}
+		});
+	}
+
+	async _handleFocusTrapEnter() {
+		this.shadowRoot.querySelector('d2l-calendar').focus(true);
```

Comment:
For the case where a user opens the calendar by clicking (focus should still remain on the `input-text`) and then tabs into the calendar, the initial focus should go to the calendar `focusDate` (which is why `true` is included)